processor/transformprocessor/config.go

```diff
@@ -129,45 +132,72 @@ func (c *Config) Unmarshal(conf *confmap.Conf) error {
 
 var _ component.Config = (*Config)(nil)
 
+func (c *Config) ValidateTraceStatements(additionalSpanFuncs ...ottl.Factory[ottlspan.TransformContext]) error {
+	var errors error
+	pc, err := common.NewTraceParserCollection(component.TelemetrySettings{Logger: zap.NewNop()}, common.WithSpanParser(traces.SpanFunctions(additionalSpanFuncs)), common.WithSpanEventParser(traces.SpanEventFunctions()))
```

Comment:
Aside from the first line, these functions are all iterating over a `common.ContextStatements` object and doing the same thing. You can refactor this into a single shared helper function.

Reply:
My initial goal while creating `Validate*Statements` was to encapsulate all reference of elements from `internal/common`, `internal/metrics`, etc, into helper functions that a developer of a custom `trasform` processor could use to create a custom `Config.Validate()`.

I'm not sure if that would work with the proposed refactoring to reduce repeated code. I removed this methods in my last iteration.